inference-engine/src/mkldnn_plugin/nodes/mkldnn_rnn.cpp

```diff
@@ -149,38 +152,52 @@ void MKLDNNRNN::fillCellDesc() {
+        IE_THROW() << "RNN Layer. Biases size is not correct. Expected size:" << G * SC;
+
+    auto runtimePrecision = outs[0]->getPrecision();
+
+    auto dataType = MKLDNNExtensionUtils::IEPrecisionToDataType(runtimePrecision);
+
+    in_states_d.resize(S);
+    out_states_d.resize(S);
 
     // Shapes and Attributes are correct. Can start internal stuff initialization.
-    for (size_t i = 0; i < S; i++) {
-        in_states_d.emplace_back(S_4D_shape, memory::data_type::f32, memory::format_tag::ldnc);
-        out_states_d.emplace_back(S_4D_shape, memory::data_type::f32, memory::format_tag::ldnc);
+    in_states_d[0]  = {S_4D_shape, dataType, memory::format_tag::ldnc};
+    out_states_d[0] = {S_4D_shape, dataType, memory::format_tag::ldnc};
+
+    if (S == 2) {
+        in_states_d[1] = {S_4D_shape, memory::data_type::f32, memory::format_tag::ldnc};
+        out_states_d[1] = {S_4D_shape, memory::data_type::f32, memory::format_tag::ldnc};
     }
 
-    in_data_d  = {{T, N, DC}, memory::data_type::f32, memory::format_tag::tnc};;
-    out_data_d = {{T, N, SC}, memory::data_type::f32, memory::format_tag::tnc};;
+    in_data_d  = {{T, N, DC}, dataType, memory::format_tag::tnc};;
+    out_data_d = {{T, N, SC}, dataType, memory::format_tag::tnc};;
 
-    w_data_d   = {{L, D, DC, G, SC}, memory::data_type::f32, memory::format_tag::ldigo};
-    w_state_d  = {{L, D, SC, G, SC}, memory::data_type::f32, memory::format_tag::ldigo};
+    w_data_d   = {{L, D, DC, G, SC}, dataType, memory::format_tag::ldigo};
+    w_state_d  = {{L, D, SC, G, SC}, dataType, memory::format_tag::ldigo};
 
     if (bias)
         w_bias_d = {{L, D, Gb, SC}, memory::data_type::f32, memory::format_tag::ldgo};
 
     std::vector<TensorDesc> in_candidate, out_candidate;
-    std::vector<memory::format_tag> outputFormats;
-    in_candidate.emplace_back(MKLDNNMemoryDesc {D_shape, memory::data_type::f32, memory::format_tag::nc});
-    in_candidate.emplace_back(MKLDNNMemoryDesc {S_shape, memory::data_type::f32, memory::format_tag::nc});
-    out_candidate.emplace_back(MKLDNNMemoryDesc {S_shape, memory::data_type::f32, memory::format_tag::nc});
-    outputFormats.emplace_back(memory::format_tag::nc);
+    in_candidate.emplace_back(MKLDNNMemoryDesc {D_shape, dataType, memory::format_tag::nc});
+    in_candidate.emplace_back(MKLDNNMemoryDesc {S_shape, dataType, memory::format_tag::nc});
+    out_candidate.emplace_back(MKLDNNMemoryDesc {S_shape, dataType, memory::format_tag::nc});
 
     if (S == 2) {
-        in_candidate.emplace_back(MKLDNNMemoryDesc {S_shape, memory::data_type::f32, memory::format_tag::nc});
-        out_candidate.emplace_back(MKLDNNMemoryDesc {S_shape, memory::data_type::f32, memory::format_tag::nc});
-        outputFormats.emplace_back(memory::format_tag::nc);
+        in_candidate.emplace_back(MKLDNNMemoryDesc {S_shape, dataType, memory::format_tag::nc});
+        out_candidate.emplace_back(MKLDNNMemoryDesc {S_shape, dataType, memory::format_tag::nc});
```

Comment:
Here the second inp/out descriptors have `dataType` precision, but in the same part of `fillSeqDesc()` function L328-329 they are initialized with `f32` data type. Could you please explain what is the difference?